### PR TITLE
feat: Platform.sh specific app pull, fixes #5727

### LINKED ISSUE #5727: Pulling data from platform.sh multiple apps
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

No

### Describe your solution

There is no way to pull data from platform.sh when you have more than one app in the environment. 

Running ddev pull platform returning this error : 
Specifying the --app or --worker is required in non-interactive mode  

It would be great to support an additional variable for the app ID. 
Something like : 
web_environment:
    - PLATFORM_PROJECT=project_id
    - PLATFORM_ENVIRONMENT=main
    - PLATFORM_APP=app ( Should be optional )

### Describe alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nir-riskified**
…an one app under your env.

## The Issue
Today there is no option to choose an app when pulling data from platform.sh using ddev pull platform

## How This PR Solves The Issue
Adding new env variable support PLATFORM_APP, when this variable exists we are adding --app flag 

## Related Issue Link(s)
https://github.com/ddev/ddev/issues/5727

### CODE REVIEW

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 1]
```diff
@@ -16,11 +16,12 @@
 #    web_environment:
 #    - PLATFORMSH_CLI_TOKEN=abcdeyourtoken
 #    ```
-# 3. Add PLATFORM_PROJECT and PLATFORM_ENVIRONMENT variables to your project `.ddev/config.yaml` or a `.ddev/config.platform.yaml`
+# 3. Add PLATFORM_PROJECT and PLATFORM_ENVIRONMENT and PLATFORM_APP(optional) variables to your project `.ddev/config.yaml` or a `.ddev/config.platform.yaml`
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:19]:
```suggestion
# 3. Add PLATFORM_PROJECT and PLATFORM_ENVIRONMENT and PLATFORM_APP (optional) variables to your project `.ddev/config.yaml` or a `.ddev/config.platform.yaml`
```

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 2]
```diff
@@ -51,7 +52,7 @@ db_pull_command:
     ls /var/www/html/.ddev >/dev/null # This just refreshes stale NFS if possible
     # /tmp/db_relationships.yaml is the full yaml output of the database relationships
     db_relationships_file=/tmp/db_relationships.yaml
-    PLATFORM_RELATIONSHIPS="" platform relationships -y  -e "${PLATFORM_ENVIRONMENT}" | yq 'with_entries(select(.[][].type == "mariadb:*" or .[][].type == "*mysql:*" or .[][].type == "postgresql:*")) ' >${db_relationships_file}
+    PLATFORM_RELATIONSHIPS="" platform relationships -y  -e "${PLATFORM_ENVIRONMENT}" $( [ -n "${PLATFORM_APP}" ] && echo "--app=${PLATFORM_APP}" ) | yq 'with_entries(select(.[][].type == "mariadb:*" or .[][].type == "*mysql:*" or .[][].type == "postgresql:*")) ' >${db_relationships_file}
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:55]:
```suggestion
    PLATFORM_RELATIONSHIPS="" platform relationships -y  -e "${PLATFORM_ENVIRONMENT}" ${PLATFORM_APP:+"--app=${PLATFORM_APP}"} | yq 'with_entries(select(.[][].type == "mariadb:*" or .[][].type == "*mysql:*" or .[][].type == "postgresql:*")) ' >${db_relationships_file}
```

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 3]
```diff
@@ -68,18 +69,18 @@ db_pull_command:
         db_name="db"
       fi
 
-      platform db:dump --yes --relationship=${rel} --gzip --file=/var/www/html/.ddev/.downloads/${db_name}.sql.gz --project="${PLATFORM_PROJECT:-setme}" --environment="${PLATFORM_ENVIRONMENT:-setme}"
+      platform db:dump --yes $( [ -n "${PLATFORM_APP}" ] && echo "--app=${PLATFORM_APP}" ) --relationship=${rel} --gzip --file=/var/www/html/.ddev/.downloads/${db_name}.sql.gz --project="${PLATFORM_PROJECT:-setme}" --environment="${PLATFORM_ENVIRONMENT:-setme}"
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:72]:
```suggestion
      platform db:dump --yes ${PLATFORM_APP:+"--app=${PLATFORM_APP}"} --relationship=${rel} --gzip --file=/var/www/html/.ddev/.downloads/${db_name}.sql.gz --project="${PLATFORM_PROJECT:-setme}" --environment="${PLATFORM_ENVIRONMENT:-setme}"
```

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 4]
```diff
@@ -68,18 +69,18 @@ db_pull_command:
...
 files_import_command:
   command: |
-    # set -x   # You can enable bash debugging output by uncommenting
+    #set -x   # You can enable bash debugging output by uncommenting
     set -eu -o pipefail
     export PLATFORMSH_CLI_NO_INTERACTION=1
     # Use $PLATFORM_MOUNTS if it exists to get list of mounts to download, otherwise just web/sites/default/files (drupal)
     declare -a mounts=(${PLATFORM_MOUNTS:-/web/sites/default/files})
-    platform mount:download --all --yes --quiet --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}"  --target=/var/www/html
+    platform mount:download --all --yes --quiet --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" $( [ -n "${PLATFORM_APP}" ] && echo "--app=${PLATFORM_APP}" ) --target=/var/www/html
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:83]:
```suggestion
    platform mount:download --all --yes --quiet --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" ${PLATFORM_APP:+"--app=${PLATFORM_APP}"} --target=/var/www/html
```

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 5]
```diff
@@ -93,7 +94,7 @@ db_push_command:
     if [ "${PLATFORM_PRIMARY_RELATIONSHIP:-}" != "" ] ; then
       rel="--relationship ${PLATFORM_PRIMARY_RELATIONSHIP}"
     fi
-    gzip -dc db.sql.gz | platform db:sql --project="${PLATFORM_PROJECT}" ${rel:-} --environment="${PLATFORM_ENVIRONMENT}"
+    gzip -dc db.sql.gz | platform db:sql --project="${PLATFORM_PROJECT}" ${rel:-} --environment="${PLATFORM_ENVIRONMENT} $( [ -n "${PLATFORM_APP}" ] && echo "--app=${PLATFORM_APP}" )"
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:97]:
```suggestion
    gzip -dc db.sql.gz | platform db:sql --project="${PLATFORM_PROJECT}" ${rel:-} --environment="${PLATFORM_ENVIRONMENT}" ${PLATFORM_APP:+"--app=${PLATFORM_APP}"}
```

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 6]
```diff
@@ -103,4 +104,4 @@ files_push_command:
     set -eu -o pipefail
     export PLATFORMSH_CLI_NO_INTERACTION=1
     ls "${DDEV_FILES_DIR}" >/dev/null # This just refreshes stale NFS if possible
-    platform mount:upload --yes --quiet --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" --source="${DDEV_FILES_DIR}" --mount=web/sites/default/files
+    platform mount:upload --yes --quiet --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" $( [ -n "${PLATFORM_APP}" ] && echo "--app=${PLATFORM_APP}" ) --source="${DDEV_FILES_DIR}" --mount=web/sites/default/files
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:107]:
```suggestion
    platform mount:upload --yes --quiet --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" ${PLATFORM_APP:+"--app=${PLATFORM_APP}"} --source="${DDEV_FILES_DIR}" --mount=web/sites/default/files
```

**Review by stasadev — CHANGES_REQUESTED** [review]
I haven't actually tested platform integration with `--app`, but I suggest a more elegant way of writing the condition.

Used `${parameter:+word}` from https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

> If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.

**Comment by stasadev** [comment]
@nir-riskified,

I checked if all changed commands have `--app` option, everything looks fine.

You can apply my suggestions in one commit on the [Files](https://github.com/ddev/ddev/pull/5728/files) tab.

And please update the [documentation file](https://github.com/ddev/ddev/blob/9d24d567065216fc1e1d85d5e2859b88a87b4728/docs/content/users/providers/platform.md) when you get a chance.

Thank you for the contribution!

File: docs/content/users/providers/platform.md [hunk 7]
```diff
@@ -22,14 +22,15 @@ web_environment:
 ## Platform.sh Per-Project Configuration
 
 1. Check out the site from Platform.sh and configure it with [`ddev config`](../usage/commands.md#config). You’ll want to use [`ddev start`](../usage/commands.md#start) and make sure the basic functionality is working.
-2. Add `PLATFORM_PROJECT` and `PLATFORM_ENVIRONMENT` variables to your project.
+2. Add `PLATFORM_PROJECT` and `PLATFORM_ENVIRONMENT` and  `PLATFORM_APP` (optional) variables to your project.
```
Comment (stasadev) [inline docs/content/users/providers/platform.md:25]:
```suggestion
2. Add `PLATFORM_PROJECT` and `PLATFORM_ENVIRONMENT` variables to your project.
```

File: docs/content/users/providers/platform.md [hunk 8]
```diff
@@ -22,14 +22,15 @@ web_environment:
...
 
     * Either in `.ddev/config.yaml` or a `.ddev/config.*.yaml` file:
 
         ```yaml
         web_environment:
         - PLATFORM_PROJECT=nf4amudfn23biyourproject
         - PLATFORM_ENVIRONMENT=main
+        - PLATFORM_APP=app (optional)
```
Comment (stasadev) [inline docs/content/users/providers/platform.md:33]:
```suggestion
```

File: docs/content/users/providers/platform.md [hunk 9]
```diff
@@ -42,6 +43,9 @@ web_environment:
 4. Run `ddev pull platform`. After you agree to the prompt, the current upstream databases and files will be downloaded.
 5. Optionally use `ddev push platform` to push local files and database to Platform.sh. The [`ddev push`](../usage/commands.md#push) command can potentially damage your production site, so we don’t recommend using it.
 
+### Multiple apps
+It is necessary to specify PLATFORM_APP variable under web_environment if your environment contains more than one app
```
Comment (stasadev) [inline docs/content/users/providers/platform.md:47]:
````suggestion
### Multiple apps

If your environment contains more than one app, add `PLATFORM_APP` variable to your project:

    * Either in `.ddev/config.yaml` or a `.ddev/config.*.yaml` file:

        ```yaml
        web_environment:
        - ...
        - PLATFORM_APP=app
        ```

    * Or with a command from your terminal:

        ```bash
        ddev config --web-environment-add="PLATFORM_APP=app"
        ```
````

File: pkg/ddevapp/dotddev_assets/providers/platform.yaml [hunk 10]
```diff
@@ -16,11 +16,12 @@
 #    web_environment:
 #    - PLATFORMSH_CLI_TOKEN=abcdeyourtoken
 #    ```
-# 3. Add PLATFORM_PROJECT and PLATFORM_ENVIRONMENT variables to your project `.ddev/config.yaml` or a `.ddev/config.platform.yaml`
+# 3. Add PLATFORM_PROJECT and PLATFORM_ENVIRONMENT and PLATFORM_APP (optional) variables to your project `.ddev/config.yaml` or a `.ddev/config.platform.yaml`
```
Comment (stasadev) [inline pkg/ddevapp/dotddev_assets/providers/platform.yaml:19]:
```suggestion
# 3. Add PLATFORM_PROJECT and PLATFORM_ENVIRONMENT and optional PLATFORM_APP (only if your environment contains more than one app) variables to your project `.ddev/config.yaml` or a `.ddev/config.platform.yaml`
```

**Review by stasadev — CHANGES_REQUESTED** [review]
@nir-riskified, I like your change in the docs with a new section for multiple apps, let's use it for information about the optional `PLATFORM_APP` instead of mixing it with other variables.

**Comment by nir-riskified** [comment]
Updated! Thanks @stasadev

**Review by stasadev — COMMENTED** [review]
Added one more commit with indentation fix in docs, and renamed the header "Multiple apps" to "Managing Multiple Apps".

Let's see how the tests go.

**Review by stasadev — APPROVED** [review]

**Pull request merged**